### PR TITLE
Fix invalid product slugs in samples

### PR DESCRIPTION
**Pull request by heaths**
At this time, there is no slug for Document Translator, so I used azure-translator-text for now.

### CODE REVIEW

**Comment by heaths** [comment]
Found using https://gist.github.com/heaths/08a395af1800a2826be8b8759e735394, which is not specific to the .NET repo.

File: sdk/translation/Azure.AI.Translation.Document/samples/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ languages:
 products:
 - azure
 - azure-cognitive-services
-- azure-document-translation
+- azure-translator-text
```
Comment (maririos) [inline sdk/translation/Azure.AI.Translation.Document/samples/README.md:8]: I think we should use `azure-translator` as it is more generic. `azure-translator-text` might only cover real-time translation, which this SDK is not targeting.
I am following with their PM to validate `azure-translator` is the right one

Comment (maririos) [reply sdk/translation/Azure.AI.Translation.Document/samples/README.md:8]: FYI @kristapratico

Comment (kristapratico) [reply sdk/translation/Azure.AI.Translation.Document/samples/README.md:8]: Agree, was going to update to azure-translator. Let us know what you learn, @maririos !

Comment (maririos) [reply sdk/translation/Azure.AI.Translation.Document/samples/README.md:8]: ok. The service team confirmed that we should use `azure-translator`

Comment (heaths) [reply sdk/translation/Azure.AI.Translation.Document/samples/README.md]: Done

**Review by maririos — APPROVED** [review]
Thank you!!!

**Review by kristapratico — APPROVED** [review]

**Pull request merged**